src/Cardano/Wallet/Kernel/Wallet.hs

```diff
@@ -0,0 +1,123 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- | Module provides internal implementation of
+-- | wallet api
+
+module Cardano.Wallet.Kernel.Wallet where
+
+import Prelude
+
+import Cardano.Wallet
+    ( Wallet )
+import Cardano.Wallet.AddressDiscovery
+    ( AddressPoolGap )
+import Cardano.Wallet.Mnemonic
+    ( Mnemonic )
+import Cardano.Wallet.Primitive
+    ( Coin )
+import Crypto.Encoding.BIP39
+    ( ValidMnemonicSentence )
+import Data.ByteArray
+    ( ScrubbedBytes )
+import Data.Text
+    ( Text )
+import Data.Time.Units
+    ( Microsecond )
+import Fmt
+    ( Buildable (..), fmt )
+
+
+-- | Errors
+data CreateWalletError = CreateWalletError
+
+instance Buildable CreateWalletError where
+    build CreateWalletError =
+        mempty
+        <> "CreateWalletError"
+
+instance Show CreateWalletError where
+    show = fmt . build
+
+data GetWalletError = GetWalletError
+
+instance Buildable GetWalletError where
+    build GetWalletError =
+        mempty
+        <> "GetWalletError"
+
+instance Show GetWalletError where
+    show = fmt . build
+
+
+-- | Types
+newtype CreateWallet = CreateWallet NewWallet
+
+data NewWallet =
+    forall mw . ValidMnemonicSentence mw => NewWallet
+    {
+      _mnemonicSentence
+        :: !(Mnemonic mw)
+    , _mnemonicSentencePassphrase
+        :: !(Maybe (Mnemonic mw))
+    , _name
+        :: !WalletName
+    , _passphrase
```

Comment:
Would this be a `Passphrase "encryption"`?
And the `mnemonicSentencePassphrase` be a `Passphrase "generation"`?
Also, are the passphrases optional? If so, what value would denote "no passphrase"?

Reply:
Replaced module defined passphrase types with address derivation ones.

I looked into swagger.yaml description of API and it says:
```
110 wallet2ndFactor: &wallet2ndFactor                                                                                                                                                         
111   description: An optional passphrase used to encrypt the mnemonic sentence.                                                                                                              
112   type: array                                                                                                                                                                             
113   minItems: 9                                                                                                                                                                             
114   maxItems: 12                                                                                                                                                                            
115   items:                                                                                                                                                                                  
116     type: string                                                                                                                                                                          
117     format: bip-0039-mnemonic-word{english}                                                                                                                                               
118   example: ["squirrel", "material", "silly", "twice", "direct", "slush", "pistol", "razor", "become"]
```